src/store/batchSlice.ts

```diff
@@ -0,0 +1,86 @@
+import { createSelector, createSlice, type PayloadAction } from '@reduxjs/toolkit'
+import type { TransactionDetails } from '@safe-global/safe-gateway-typescript-sdk'
+import type { RootState } from '.'
+
+export type DraftBatchItem = {
+  id: string
+  timestamp: number
+  txDetails: TransactionDetails
+}
+
+type BatchTxsState = {
+  [chainId: string]: {
+    [safeAddress: string]: DraftBatchItem[]
+  }
+}
+
+const initialState: BatchTxsState = {}
+
+export const batchSlice = createSlice({
+  name: 'batch',
+  initialState,
+  reducers: {
+    // Set a batch (used for reordering)
+    setBatch: (
+      state,
+      action: PayloadAction<{
+        chainId: string
+        safeAddress: string
+        items: DraftBatchItem[]
+      }>,
+    ) => {
+      const { chainId, safeAddress, items } = action.payload
+      state[chainId] = state[chainId] || {}
+      // @ts-ignore
+      state[chainId][safeAddress] = items
+    },
+
+    // Add a tx to the batch
+    addTx: (
+      state,
+      action: PayloadAction<{
+        chainId: string
+        safeAddress: string
+        txDetails: TransactionDetails
+      }>,
+    ) => {
+      const { chainId, safeAddress, txDetails } = action.payload
+      state[chainId] = state[chainId] || {}
+      state[chainId][safeAddress] = state[chainId][safeAddress] || []
+      // @ts-ignore
```

Comment:
Same here :)

Reply:
Removed this one, the other one seems to be needed.